app/jobs/search_keyword_job.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class SearchKeywordJob < ApplicationJob
+  def perform(keyword_id)
+    keyword = Keyword.includes(:source).find(keyword_id)
+    keyword_name = keyword.name
+    search_result = search_service(keyword).new(keyword_name).call
+    update_keyword(keyword, search_result)
+  end
+
+  private
+
+  def search_service(keyword)
+    source_name = keyword.source.name.downcase
+    return Bing::SearchService if source_name == 'bing'
```

Comment:
In Rails, when comparing the string like this, we prefer to use symbol to save the memory. https://stackoverflow.com/questions/16621073/when-to-use-symbols-instead-of-strings-in-ruby

```suggestion
    source_name = keyword.source.name.to_sym
    return Bing::SearchService if source_name == :bing
```

https://stackoverflow.com/questions/16621073/when-to-use-symbols-instead-of-strings-in-ruby

Reply:
@byhbt I would be careful in this case as the model does not have any validations on "name" to ensure is "safe" to convert and compare as a symbol 🤔 
Symbols are better for enum strings, constants, param/attributes names, ... as these are fully in the control of the code. The source name might be a string input from an API.

Reply:
right, also make sense.